pkg/router/template/router.go

```diff
@@ -86,13 +85,23 @@ type templateRouter struct {
 	statsPort int
 	// if the router should allow wildcard routes.
 	allowWildcardRoutes bool
-	// rateLimitedCommitFunction is a rate limited commit (persist state + refresh the backend)
-	// function that coalesces and controls how often the router is reloaded.
-	rateLimitedCommitFunction *ratelimiter.RateLimitedFunction
-	// rateLimitedCommitStopChannel is the stop/terminate channel.
-	rateLimitedCommitStopChannel chan struct{}
-	// lock is a mutex used to prevent concurrent router reloads.
+	// lock is a mutex used to prevent concurrent router state updates
 	lock sync.Mutex
+	// Track the start and end of router reloads
+	lastReloadStart time.Time
+	lastReloadEnd   time.Time
+	// commitReqTime is nil if no commit is needed, otherwise it is the time the commit was requested
+	commitReqTime *time.Time
+	// commitRunning indicates whether there is the commitFunc is actively running
+	commitRunning bool
+	// commitTimer is the timer we use to make callbacks to the delayed commits
+	commitTimer *time.Timer
+	// The minimum time between the starts of reloads
```

Comment:
godoc: `reloadInterval is the minimum...`

Reply:
ditto the next few fields

Reply:
Done